File: sitesync-angular/src/app/core/interceptors/models/auth.models.ts
// src/app/core/models/auth.models.ts

export type UserRole = 'Admin' | 'Supervisor' | 'Labour';

export interface LoginRequest {
  username: string;
  password: string;
}

export interface LoginResponse {
  token:    string;
  username: string;
  role:     UserRole;
  fullName: string;
  expires:  string;
}

export interface AuthUser {
  username: string;
  fullName: string;
  role:     UserRole;
  token:    string;
  expires:  string;
}

// What each role can access
export const ROLE_PERMISSIONS: Record<UserRole, string[]> = {
  Admin:      ['dashboard','sites','clients','labour','finance','analytics','settings'],
  Supervisor: ['dashboard','sites','labour','analytics'],
  Labour:     ['dashboard','labour'],
};

File: sitesync api/SiteSync.API/Models/AuthModels.cs
// SiteSync.API/Models/AuthModels.cs
namespace SiteSync.API.Models;

public class LoginRequest
{
    public string Username { get; set; } = string.Empty;
    public string Password { get; set; } = string.Empty;
}

public class LoginResponse
{
    public string Token { get; set; } = string.Empty;
    public string Username { get; set; } = string.Empty;
    public string Role { get; set; } = string.Empty;
    public string FullName { get; set; } = string.Empty;
    public DateTime Expires { get; set; }
}

public class AppUser
{
    public int Id { get; set; }
    public string Username { get; set; } = string.Empty;
    public string Password { get; set; } = string.Empty;  // mapped from PasswordHash column
    public string FullName { get; set; } = string.Empty;
    public string Role { get; set; } = string.Empty;
    public bool IsActive { get; set; } = true;
}